### PR TITLE
fix #361 ;;adebug will return message if LL is enabled

### LINKED ISSUE #361: ;;adebug not work as intended
We are now using LavaLink, so this command might not work/borked.

### PR DESCRIPTION
**Pull request by knight-ryu12**
Ref #361

### CODE REVIEW

File: FredBoat/src/main/java/fredboat/command/maintenance/AudioDebugCommand.java [hunk 1]
```diff
@@ -44,7 +45,15 @@ public AudioDebugCommand(String name, String... aliases) {
 
     @Override
     public void onInvoke(@Nonnull CommandContext context) {
-        String msg = "";
+        if (LavalinkManager.ins.isEnabled()) {
+            //handleLavalink(context);
+            context.replyWithName("Lavalink is enabled");
```
Comment (napstr) [inline FredBoat/src/main/java/fredboat/command/maintenance/AudioDebugCommand.java]: This looks not like a helpful message. A hint where to find the stats for lavalink would be better (nodes command).

File: FredBoat/src/main/java/fredboat/command/maintenance/AudioDebugCommand.java [hunk 2]
```diff
@@ -59,9 +68,27 @@ public void onInvoke(@Nonnull CommandContext context) {
         }
 
         context.replyWithName(msg);
```
Comment (napstr) [inline FredBoat/src/main/java/fredboat/command/maintenance/AudioDebugCommand.java]: Why keep this around?

**Comment by knight-ryu12** [comment]
Changed.

File: FredBoat/src/main/java/fredboat/command/maintenance/AudioDebugCommand.java [hunk 3]
```diff
@@ -44,6 +45,18 @@ public AudioDebugCommand(String name, String... aliases) {
...
     public void onInvoke(@Nonnull CommandContext context) {
+        if (LavalinkManager.ins.isEnabled()) {
+            handleLavalink(context);
+        } else {
+            handleLavaplayer(context);
+        }
+    }
+
+    private void handleLavalink(CommandContext context) {
+        context.replyWithName("LavaLink is enabled! please use ;;debug instead.");
```
Comment (napstr) [inline FredBoat/src/main/java/fredboat/command/maintenance/AudioDebugCommand.java]: Hardcoding the prefix is meh. Please find another way. Also ;;debug does not show the equivalent of the information that ;;adebug is meant to show. See my previous command about which command does that.

**Comment by knight-ryu12** [comment]
Do we need to check args of this command?
Nodes does check args to check which node is selected

File: FredBoat/src/main/java/fredboat/command/maintenance/AudioDebugCommand.java [hunk 4]
```diff
@@ -44,6 +50,19 @@ public AudioDebugCommand(String name, String... aliases) {
...
     public void onInvoke(@Nonnull CommandContext context) {
+        if (LavalinkManager.ins.isEnabled()) {
+            handleLavalink(context);
+        } else {
+            handleLavaplayer(context);
+        }
+    }
+
+    private void handleLavalink(CommandContext context) {
+        context.replyWithName("LavaLink is enabled! showing LavaLink status.");
```
Comment (napstr) [inline FredBoat/src/main/java/fredboat/command/maintenance/AudioDebugCommand.java]: Please start sentences with an upper case letter.

**Comment by knight-ryu12** [comment]
English issue has been fixed + slightly changed

File: FredBoat/src/main/java/fredboat/command/maintenance/AudioDebugCommand.java [hunk 5]
```diff
@@ -27,14 +27,20 @@
...
 import fredboat.commandmeta.abs.IMaintenanceCommand;
 import fredboat.messaging.internal.Context;
 import fredboat.util.TextUtils;
+import lavalink.client.io.Lavalink;
+import lavalink.client.io.LavalinkSocket;
+import lavalink.client.io.RemoteStats;
 
 import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
```
Comment (Shredder121) [inline FredBoat/src/main/java/fredboat/command/maintenance/AudioDebugCommand.java]: I think you're importing quite a few classes that you aren't using.

Comment (knight-ryu12) [reply FredBoat/src/main/java/fredboat/command/maintenance/AudioDebugCommand.java]: oh. totally forgot that

**Comment by knight-ryu12** [comment]
lgtm? owo

2017/10/28 午前8:23 "Napster" <notifications@github.com>:

**Pull request merged**